### PR TITLE
[cli] asset materialize non zero exit code on failure

### PR DESCRIPTION
**Pull request by alangenfeld**
follow the pattern we use for `job execute` 
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_cli/job.py#L348

resolves https://github.com/dagster-io/dagster/issues/19009

## How I Tested These Changes

need to add a test

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19029** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19029?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19029?utm_source=stack-comment).

File: python_modules/dagster/dagster/_cli/asset.py [hunk 1]
```diff
@@ -78,9 +78,11 @@ def execute_materialize_command(instance: DagsterInstance, kwargs: Mapping[str,
     else:
         tags = {}
 
-    execute_job(
+    result = execute_job(
         job=reconstructable_job, asset_selection=list(asset_keys), instance=instance, tags=tags
     )
+    if not result.success:
+        raise click.ClickException("Materialization failed.")
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_cli/asset.py:85]: is there a way to know what asset failed and report it here?

**Review by jamiedemaria — APPROVED** [review]
lgtm

File: python_modules/dagster/dagster/_cli/asset.py (hunk 1 above)
Comment (alangenfeld) [reply python_modules/dagster/dagster/_cli/asset.py:85]: likely, though the output of running the CLI contains the events logged out which has the full context of failure already 

example here: https://gist.github.com/alangenfeld/04639f6ebaa7bcbc962bd6028731ebbf

Comment (jamiedemaria) [reply python_modules/dagster/dagster/_cli/asset.py:85]: ok! does't seem worth it then

**Pull request merged**